File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l,r = 0,len(heights)-1
        res = 0
        while l<r :
            height = min(heights[l],heights[r])
            area = height*(r-l)
            res = max(res,area)
            if heights[l] < heights[r]:
                l+=1
            else:
                r-=1
        return res

File: Data Structures & Algorithms/max-area-of-island/submission-0.cpp
class Solution {
public:
    int maxAreaOfIsland(vector<vector<int>>& grid) {
        int res = 0;
        for(int i = 0; i < grid.size(); i++){
            for(int j = 0; j < grid[0].size(); j++){
                if(grid[i][j] == 1){
                    res = max(helper(grid,i,j),res);
                }             
            }
        }
        return res;
    }
private:
    int helper(vector<vector<int>>& grid, int r, int c){
        if(r < 0 || c < 0 || r >= grid.size() || c >= grid[0].size() || grid[r][c] == 0){
            return 0;
        }
        int area = 1;
        grid[r][c] = 0;
        area += helper(grid,r-1,c);
        area += helper(grid,r+1,c);
        area += helper(grid,r,c-1);
        area += helper(grid,r,c+1);
        return area;

    }
};
